### PR TITLE
get correct projecttype or role if missing

### PR DESCRIPTION
**Pull request by sarahentzel**

### CODE REVIEW

File: src/crud/useProjectType.ts [hunk 1]
```diff
@@ -1,38 +1,60 @@
...
     if (ptId) {
-    var pt = memory.cache.query((q) =>
+      pt = memory.cache.query((q) =>
         q.findRecord({ type: 'projecttype', id: ptId })
       ) as ProjectType;
       return pt.attributes.name;
-    }
-    else
+    } else {
       console.log('MISSING PROJECT TYPE!', proj);
```
Comment (gtryus) [inline src/crud/useProjectType.ts:29]: If we think this might happen, do we want to report a bug snag error?

File: src/crud/useProjectType.ts [hunk 2]
```diff
@@ -1,38 +1,60 @@
...
-    return '';
-};
+      var pts = memory.cache.query((q) =>
+        q.findRecords('projecttype')
+      ) as ProjectType[];
+      var online = Boolean(proj.keys?.remoteId);
+      pts = pts.filter(
+        (p) =>
+          Boolean(p.keys?.remoteId) === online &&
+          p.attributes.name === 'Scripture'
```
Comment (gtryus) [inline src/crud/useProjectType.ts:37]: Do we want a comment that we default to Scripture if no type is found?

File: src/crud/useRole.ts [hunk 3]
```diff
@@ -66,10 +66,20 @@ export const useRole = () => {
 
   const getMbrRole = (memberRecs: Record[]) => {
     if (memberRecs.length === 1) {
-      const roleId = related(memberRecs[0], 'role');
+      var roleId = related(memberRecs[0], 'role');
+      if (!roleId) {
+        roleId = getRoleId(RoleNames.Admin);
```
Comment (gtryus) [inline src/crud/useRole.ts:71]: Would we want to default to Editor if no role is found?

**Review by gtryus — COMMENTED** [review]
I left some comments...

File: src/crud/useRole.ts (hunk 3 above)
Comment (sarahentzel) [reply src/crud/useRole.ts:71]: No, because this happens when we create a project, so we'd want to be admin

**Review by gtryus — APPROVED** [review]

**Pull request merged**